src/app/mydata/services/collaborations.service.ts

```diff
@@ -32,15 +32,15 @@ export class CollaborationsService {
   }
 
   updateTokenInHttpAuthHeader() {
-    const token = this.oidcSecurityService.getAccessToken();
-
-    this.httpOptions = {
-      headers: new HttpHeaders({
-        'Content-Type': 'application/json',
-        Authorization: `Bearer ${token}`,
-      }),
-      observe: 'response',
-    };
+    this.oidcSecurityService.getAccessToken().subscribe((token) => {
```

Comment:
1. The subscription should be passed a Observer instead of a function

{
  next: (value: T) => void;
  error: (err: any) => void;
  complete: () => void;
}

Passing functions will be deprecated in the future.

2. Please confirm that the subscription completes and a memory leak is avoided.